runtime/gcp/API Gateway/gcp_api_key_restrictions.ts

```diff
@@ -0,0 +1,92 @@
+import { ApiKeysClient } from "@google-cloud/apikeys";
+import { printSummary, generateSummary } from "../../utils/string-utils.js";
+import { ComplianceStatus, type ComplianceReport, type RuntimeTest } from "../../types.js";
+
+// Helper function to check API key restrictions
+function hasApiRestrictions(key: any): boolean {
+	// Check if key has restrictions and at least one API target
+	return (
+		key.restrictions &&
+		Array.isArray(key.restrictions) &&
+		key.restrictions.length > 0 &&
+		key.restrictions.some(
+			(restriction: any) =>
+				restriction.apiTargets &&
+				Array.isArray(restriction.apiTargets) &&
+				restriction.apiTargets.length > 0
+		)
+	);
+}
+
+// Main compliance check function
+export async function checkApiKeyRestrictions(
+	projectId: string = process.env.GCP_PROJECT_ID || ""
+): Promise<ComplianceReport> {
+	const client = new ApiKeysClient();
+	const results: ComplianceReport = {
+		checks: []
+	};
+
+	try {
+		// List API keys for the project
+		const [keys] = await client.listKeys({
```

Comment:
get all by pagination, otherwise it will only process first 50

Reply:
done